### ISSUE TITLE
Support $(uname -m) on micromamba.snakepit.net

### ISSUE THREAD

**Issue by maresb** [mamba-org/mamba#1437]
In the context of micromamba-docker and jupyter/docker-stacks, it would be really convenient to be able to run 

```bash
wget -qO- https://micromamba.snakepit.net/api/micromamba/$(uname)-$(uname -m)/latest | tar -xvj bin/micromamba
```

instead of

```bash
ARCH="Bash is annoying"
wget -qO- https://micromamba.snakepit.net/api/micromamba/linux-$ARCH/latest | tar -xvj bin/micromamba
```

The problem is that the Conda name (since micromamba's archive comes from [anaconda.org](https://anaconda.org/conda-forge/micromamba/files)) disagrees with the name provided by `uname -m`. As reference, [Micromamba](https://github.com/conda-forge/miniforge/releases) names their releases according to the `uname -m` convention.

Anyways, it's really painful to mess around with dictionaries when working with primitive tools like bash. It'd be great if the translation could occur in snakepit where you are presumably running something slick like FastAPI.

I think the dictionary you'll want to use is:

`Linux-x86_64` → `linux-64`
`Linux-aarch64` → `linux-aarch64`
`Linux-ppc64le` → `linux-ppc64le`
`MacOSX-x86_64` → `osx-64`
`MacOSX-arm64` → `osx-arm64`
`Windows-x86_64` → `win-64`

**Comment by jonashaag** [comment]
On my M1 MacBook:

```
echo $(uname)-$(uname -m)
Darwin-arm64
```

Shortest one liner I could come up with:

```
bash -c 'x=$(uname)-$(uname -m) x=${x,,} x=${x//darwin/osx} x=${x//_x86/}; echo $x'
```

**Comment by wolfv** [comment]
I think it could be nice to make a canonical bash/zsh script to download micromamba (and also check sha256sum etc).

This could be used like `curl micro.mamba.pm | bash` and would automatically figure out the architecture etc.

**Comment by maresb** [comment]
Thanks @jonashaag for catching that, I have appended two Darwin entries to my dictionary.

I'm not sure I like the idea of canonical downloadable scripts, the reason being that one is typically working on a very stripped-down environment, so there is some dependency management which must take place beforehand, namely installing `bzip2` and `ca-certificates` when in Ubuntu. It seems better to keep that explicit rather than hide that away in a script. I actually really like the way it's currently being done with the suggested one-liner and the webservice redirect. Or do you have some clever idea for solving everything with bash? (Also, who is the authority in this case for the sha256?)

On a side note, regarding the one-liner, I would advocate adding `--strip-components=1` to the tar command so that we extract `micromamba` instead of `bin/micromamba`.

**Comment by jonashaag** [comment]
Minimal implementation:

```
curl https://raw.githubusercontent.com/jonashaag/micromamba-installer/master/install.sh | bash -s m2
```

Can also pin a version for security reasons:

```
curl https://raw.githubusercontent.com/jonashaag/micromamba-installer/2b621471e5203d60dc9838572a9e012e77e085e0/install.sh | bash -s m2
```

**Comment by jonashaag** [comment]
I agree with you @maresb. The script essentially is a wrapper for the one-liner and the other stuff it's doing isn't necessary. Unless we add more features to the script (what features?) I don't see the point either and agree that we should add this to the web service.

Btw, is there a trick to figure out the OS from a web request? That would be really cool, then we could skip the `uname` part (which isn't available on a plain Windows installation).

**Comment by maresb** [comment]
> Btw, is there a trick to figure out the OS from a web request? That would be really cool, then we could skip the uname part (which isn't available on a plain Windows installation).

@jonashaag, that's a really good question. I believe the only info we'd have is from the request headers.

Giving it a try with `wget --debug`,
```
wget --debug -qO /dev/null "https://micromamba.snakepit.net/api/micromamba/linux-64/latest"

...

---request begin---
GET /api/micromamba/linux-64/latest HTTP/1.1
User-Agent: Wget/1.20.3 (linux-gnu)
Accept: */*
Accept-Encoding: identity
Host: micromamba.snakepit.net
Connection: Keep-Alive

---request end---

...
```

so in this case we have `linux-gnu`.

Trying with `curl -v`,

```
curl -v "https://micromamba.snakepit.net/api/micromamba/linux-64/latest" > /dev/null

...

> GET /api/micromamba/linux-64/latest HTTP/1.1
> Host: micromamba.snakepit.net
> User-Agent: curl/7.68.0
> Accept: */*

...
```

So in this case we have no info. It seems the answer is "sometimes, but not reliably."
